sdk/core/core/src/az_span.c

```diff
@@ -287,44 +340,31 @@ az_span_copy_url_encode(az_span destination, az_span source, az_span* out_span)
       s += 3;
     }
   }
-  *out_span = az_span_init(az_span_ptr(destination), s, az_span_capacity(destination));
+  *out_span = az_span_slice(destination, result_size, -1);
 
   return AZ_OK;
 }
 
 AZ_NODISCARD az_result
 az_span_to_str(char* destination, int32_t destination_max_size, az_span source)
 {
+  AZ_PRECONDITION_NOT_NULL(destination);
+  AZ_PRECONDITION(destination_max_size > 0);
   AZ_PRECONDITION_VALID_SPAN(source, 0, true);
 
-  int32_t span_length = az_span_length(source);
-  if (span_length + 1 > destination_max_size)
+  int32_t span_size = az_span_size(source);
+  if (destination_max_size < 0 || (uint32_t)(span_size + 1) > (uint32_t)destination_max_size)
```

Comment:
I'll add a comment. This is similar to some of the other places where we try to guard against AV/memory corruption in cases that precondition is turned off. 

If someone were to pass in negative `destination_max_size` with precondition turned off, then the `span_size + 1` check would fail to catch that and now we are potentially copying the data from the span to an arbitrary pointer. This would definitely be a user bug, and maybe we can remove it.

I would also be OK with removing such extra checks across the board, given we are working in C. :)

You bring up a good point though, `destination_max_size` should also be a precondition.

Reply:
My point was that it looks like you added the precondition and then with that update, maybe this should be removed?